### PR TITLE
feat: convert

### PR DESCRIPTION
**Pull request by nchamo**
We are now implementing `convert`. This function will:
- Take the batch's funds
- Wrap the tokens (if needed)
- Deposit the funds to the hub
- Associate the new position to the given nonce
- Pay the subsidies (if applies)

_Note: I've only added unit tests that verify revert flows. Will add a good E2E test in another PR_

### CODE REVIEW

File: src/bridges/mean/MeanBridge.sol [hunk 1]
```diff
@@ -57,6 +58,43 @@ contract MeanBridge is BridgeBase, Ownable2Step {
...
+        uint256 _inputValue,
+        uint256 _interactionNonce,
+        uint64 _auxData,
+        address _rollupBeneficiary
+    ) external payable override(BridgeBase) onlyRollup returns (uint256, uint256, bool) {
+        if (_inputAssetA.assetType != _outputAssetB.assetType || _inputAssetA.erc20Address != _outputAssetB.erc20Address) {
+            // We are making sure that input asset A = output asset B. We do this so that, if we need to, we can close the 
+            // position while it's still being swapped
+            revert ErrorLib.InvalidOutputB();
+        }
```
Comment (0xged) [inline src/bridges/mean/MeanBridge.sol:85]: Do we need to check if inputA != outputA? I know our smart contract on Ethereum mainnet will, but no sure how will that work for the Aztec transaction. Just double-checking with this Q tbh.

File: src/bridges/mean/MeanBridge.sol [hunk 2]
```diff
@@ -151,7 +189,51 @@ contract MeanBridge is BridgeBase, Ownable2Step {
...
+        );
+
+        // Compute the criteria for this position
+        _criteria = computeCriteriaForPosition(_from, _to, _amountOfSwaps, _swapInterval);
+    }
+
+    function _wrapIfNeeded(AztecTypes.AztecAsset memory _inputAsset, address _hubToken, uint256 _amountToWrap) internal returns (uint256 _wrappedAmount) {
+        if (_inputAsset.assetType == AztecTypes.AztecAssetType.ETH) {
+            WETH.deposit{value: _amountToWrap}();            
+        } else if (_inputAsset.erc20Address != _hubToken) {
```
Comment (0xged) [inline src/bridges/mean/MeanBridge.sol]: So, if `inputAsset` is DAI, then `_hubToken` here will be `weDAI`, right? smort 🧠

**Review by 0xged — APPROVED** [review]
looks gucci thus far!

File: src/bridges/mean/MeanBridge.sol (hunk 2 above)
Comment (nchamo) [reply src/bridges/mean/MeanBridge.sol]: Yes sir!

File: src/bridges/mean/MeanBridge.sol (hunk 1 above)
Comment (nchamo) [reply src/bridges/mean/MeanBridge.sol:85]: I don't think it's necessary since the whole tx would fail when the hub reverts

**Pull request merged**